js/client.js

```diff
@@ -935,7 +935,7 @@
 				var nlIndex = data.indexOf('\n');
 				if (nlIndex > 0) {
 					this.receive(data.substr(nlIndex + 1));
-					parts = data.substr(0, nlIndex).split('|');
+					parts = data.substr(1, nlIndex).split('|');
```

Comment:
```suggestion
					parts = data.slice(1, nlIndex).split('|');
```

Remember, `substr(start, length)`, `slice(start, end)`

Reply:
They technically mean the same thing here, but yeah `slice` is clearer`.

Reply:
No, they don't mean the same thing, that's why I said **length** vs **end**.